### PR TITLE
HttpWebRequestClient is not open for extensions anymore

### PR DESCRIPTION
**Pull request by Mpdreamz**
Making it impossible for users to subclass the implementation to for instance add in Kerberos authentication.

### CODE REVIEW

File: src/Elastic.Transport/Components/TransportClient/HttpWebRequestTransportClient.cs [hunk 1]
```diff
@@ -212,7 +213,7 @@ public override TResponse Request<TResponse>(RequestData requestData)
 	/// Allows subclasses to modify the <see cref="HttpWebRequest"/> instance that is going to be used for the API call
 	/// </summary>
 	/// <param name="requestData">An instance of <see cref="RequestData"/> describing where and how to call out to</param>
-	internal HttpWebRequest CreateHttpWebRequest(RequestData requestData)
+	protected virtual HttpWebRequest CreateHttpWebRequest(RequestData requestData)
```
Comment (swallez) [inline src/Elastic.Transport/Components/TransportClient/HttpWebRequestTransportClient.cs:216]: Should this one be open for extension? Isn't allowing extensions on `SetAuthenticationIfNeeded` enough?

**Review by technige — APPROVED** [review]
LGTM :+1:

File: src/Elastic.Transport/Components/TransportClient/HttpWebRequestTransportClient.cs (hunk 1 above)
Comment (Mpdreamz) [reply src/Elastic.Transport/Components/TransportClient/HttpWebRequestTransportClient.cs:216]: For this specific use case yes, This was always open for extension to to anything bespoke where the virtual calls inside of the default implementation acted as unenforced encapsulation. 

E.g nothing enforces `SetAuthenticationIfNeeded` implementations only touch authentication but if you want to take full ownership of the creation of the HttpWebRequest and do something more bespoke it makes sense to subclass this instead and possibly invoke `base.CreateHttpWebRequest()` in the base class implementation.

This is how it used to be in < 8.x and the comment still matches the restored behaviour.

**Review by swallez — APPROVED** [review]
LGTM.

**Pull request merged**